tls/s2n_connection.h

```diff
@@ -123,6 +123,13 @@ struct s2n_connection {
     /* Connection successfully set a ticket on the connection */
     unsigned set_session:1;
 
+    /*
+     * This option exists to allow for polling the client_hello callback.
+     *
+     * Note: This defaults to false to ensure backwards compatibility.
+     */
+    unsigned client_hello_callback_poll:1;
```

Comment:
Why isn't this on the client_hello struct?

Reply:
the client_hello struct gets overwritten in `parse_client_hello` and the value is overwritten. Now there might be a way to set it after we do the parsing or more subtly but that seems more confusing to reason about.

Reply:
What if that function worked similar to the CH cb done where the callback actually enables it. This would be done post-parse so the overwriting wouldn't be an issue.